src/blockdata/constants.rs

```diff
@@ -250,5 +277,54 @@ mod test {
         assert_eq!(format!("{:x}", gen.header.block_hash()),
                    "00000008819873e925422c1ff0f99f7cc9bbb232af63a077a480a3633bee1ef6".to_string());
     }
+
+    // The *_chain_hash tests are sanity/regression tests, they verify that the const byte array
+    // representing the genesis block is the same as that created by hashing the genesis block.
+    fn chain_hash_and_genesis_block(network: Network) -> (String, String) {
+        use hashes::{sha256, Hash};
+
+        // The genesis block hash is a double-sha256 and it is displayed backwards.
+        let genesis_hash = genesis_block(network).block_hash();
+        // We abuse the sha256 hash here so we get a LowerHex impl that does not print the hex backwards.
+        let hash = sha256::Hash::from_slice(&genesis_hash.into_inner()).unwrap();
+        let want = format!("{:02x}", hash);
+
+        let chain_hash = ChainHash::using_genesis_block(network);
+        let got = format!("{:02x}", chain_hash);
+
+        (got, want)
+    }
+
+    #[test]
+    fn mainnet_chain_hash_genesis_block() {
+        let (got, want) = chain_hash_and_genesis_block(Network::Bitcoin);
+        assert_eq!(got, want);
+    }
+
+    #[test]
+    fn testnet_chain_hash_genesis_block() {
+        let (got, want) = chain_hash_and_genesis_block(Network::Testnet);
+        assert_eq!(got, want);
+    }
+
+    #[test]
+    fn signet_chain_hash_genesis_block() {
+        let (got, want) = chain_hash_and_genesis_block(Network::Signet);
+        assert_eq!(got, want);
+    }
+
+    #[test]
+    fn regtest_chain_hash_genesis_block() {
+        let (got, want) = chain_hash_and_genesis_block(Network::Regtest);
+        assert_eq!(got, want);
+    }
```

Comment:
Use a macro to deduplicate? And perhaps move `assert_eq` to the common function.

Reply:
Done, cheers!